common/ts/__tests__/publish-task-test.ts

```diff
@@ -0,0 +1,119 @@
+import Analysis from '../sonarqube/Analysis';
```

Comment:
Most of these tests are for the pre-existing behaviour. I'll call out the check that covers the change made in the fix.